app/post.go

```diff
@@ -506,6 +507,11 @@ func (a *App) UpdatePost(post *model.Post, safeUpdate bool) (*model.Post, *model
 		newPost.Props = post.Props
 	}
 
+	// Avoid deep-equal checks if EditAt was already modified through message change
+	if newPost.EditAt == oldPost.EditAt && (!reflect.DeepEqual(oldPost.FileIds, newPost.FileIds) || !reflect.DeepEqual(oldPost.Attachments(), newPost.Attachments())) {
```

Comment:
The use of DeepEqual makes me a bit nervous here. Can we maybe use a custom comparator?

Reply:
Yeah I think that's a good call, although it introduced ~448 LoC of code for both comparison operators + tests. I think since they are reusable and don't rely on reflection, that's a better way to go.